actions/integrate/action.yml

```diff
@@ -188,126 +188,60 @@ runs:
           }
 
-          let resultTable = `| File name | Link | Result count | Errors |\n| --- | --- | --- | --- |\n`;
-
-          for (const [key, value] of Object.entries(testResults)) {
-            for (const result of value) {
-              resultTable += `| ${key} | [See in Explore](${result.link}) | ${result.stats.count} | ${result.stats.errors.length} |\n`;
-            }
-          }
-
-          const prData = await github.rest.pulls.get({
-            owner: context.repo.owner,
-            repo: context.repo.repo,
-            pull_number: process.env.PULL_REQUEST_NUMBER,
-          });
-          if (!prData.data) {
-            console.log(`No pull request found for ${context.repo.owner}/${context.repo.repo}#${process.env.PULL_REQUEST_NUMBER}`);
-            return '';
-          }
-
-          const nodeId = prData.data.node_id;
-
-          const comment = `
-          ### Sigma Rule Integrations
-
-          | Changed | Deleted |
-          | --- | --- |
-          | ${changedFiles.length} | ${deletedFiles.length} |
-
-          ### Changed Files
-
-          ${changedFiles.length ? changedFiles.map(file => `- [${file}](https://github.com/${context.repo.owner}/${context.repo.repo}/blob/${prData.data.head.ref}/${file})`).join("\n") : "No files changed"}
-
-          ### Deleted Files
-
-          ${deletedFiles.length ? deletedFiles.map(file => `- ${file}`).join("\n") : "No files deleted"}
-
-          ### Test Results
-
-          ${resultTable}
-          `;
-
-          const oldCommentQuery = `query GetPRComments($owner: String!, $name: String!, $number: Int!) {
-              repository(owner: $owner, name: $name) {
-                id
-                pullRequest(number: $number) {
-                  title
-                  comments(last: 100) {
-                    nodes {
-                      id
-                      bodyText
-                      isMinimized
-                      author {
-                        login
-                      }
-                    }
-                  }
-                }
-              }
-          }`;
-
-          const minimizeCommentMutation = `mutation MinimizeComment($subjectId: ID!) {
-            minimizeComment(input: {
-              subjectId: $subjectId,
-              classifier: OUTDATED
-            }) {
-              clientMutationId
-            }
-          }`;
-
-          const query = `mutation AddComment($body: String!, $subjectId: ID!) {
-            addComment(input: {
-              body: $body,
-              subjectId: $subjectId,
-            }) {
-              subject {
-                id
-                ... on PullRequest {
-                  number
-                }
+          // Function to extract title from JSON file
+          function extractTitle(filePath) {
+            try {
+              const content = fs.readFileSync(filePath, 'utf8');
+              const titleMatch = content.match(/"title":\s*"([^"]+)"/);
```

Comment:
I assume this is for performance, but I'm personally not a huge fan of using regexes to parse JSON. I am sure this works correctly, but I fear there are edge cases it won't work well for - like unusual characters in the title. What would the performance impact of using a JSON parser here be?